ui/selectors/selectors.js

```diff
@@ -592,6 +592,23 @@ export function getSelectedAccountNativeTokenCachedBalanceByChainId(state) {
   return balancesByChainId;
 }
 
+export const getSelectedAccountNativeTokenCachedBalanceByChainIdDeepEq =
+  createDeepEqualSelector(
+    (state) => state.metamask.accountsByChainId,
+    getSelectedInternalAccount,
+    (accountsByChainId, { address: selectedAddress }) => {
+      const balancesByChainId = {};
+      for (const [chainId, accounts] of Object.entries(
+        accountsByChainId || {},
+      )) {
+        if (accounts[selectedAddress]) {
+          balancesByChainId[chainId] = accounts[selectedAddress].balance;
+        }
+      }
+      return balancesByChainId;
+    },
+  );
+
 /**
```

Comment:
This differs from cherry-pick as we have done a large refactor
https://github.com/MetaMask/metamask-extension/pull/29886